src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsintegrationapi/services/GetPersonService.kt

```diff
@@ -2,17 +2,41 @@ package uk.gov.justice.digital.hmpps.hmppsintegrationapi.services
 
 import org.springframework.beans.factory.annotation.Autowired
 import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.hmppsintegrationapi.gateways.PrisonerOffenderSearchGateway
 import uk.gov.justice.digital.hmpps.hmppsintegrationapi.gateways.ProbationOffenderSearchGateway
 import uk.gov.justice.digital.hmpps.hmppsintegrationapi.models.hmpps.Person
 import uk.gov.justice.digital.hmpps.hmppsintegrationapi.models.hmpps.Response
 
 @Service
 class GetPersonService(
+  @Autowired val prisonerOffenderSearchGateway: PrisonerOffenderSearchGateway,
   @Autowired val probationOffenderSearchGateway: ProbationOffenderSearchGateway,
 ) {
-  fun execute(hmppsId: String): Response<Person?> {
-    val personFromProbationOffenderSearch = probationOffenderSearchGateway.getPerson(id = hmppsId)
+  fun execute(hmppsId: String): Response<Map<String, Person?>> {
+    val probationResponse = probationOffenderSearchGateway.getPerson(id = hmppsId)
 
-    return Response(data = personFromProbationOffenderSearch.data, errors = personFromProbationOffenderSearch.errors)
+    val prisonResponse =
+      probationResponse.data?.identifiers?.nomisNumber?.let {
+        prisonerOffenderSearchGateway.getPrisonOffender(nomsNumber = it)
+      }
+
+    return if (prisonResponse != null) {
+      Response(
+        data =
+          mapOf(
+            "prisonerOffenderSearch" to prisonResponse.data?.toPerson(),
+            "probationOffenderSearch" to probationResponse.data,
+          ),
+        errors = prisonResponse.errors + probationResponse.errors,
```

Comment:
I'm worried that this may not be correct, as everywhere else in the code I have seen, if one of the responses has errors then we return the errors only and no data for that response, but here we seem to be returning both data and errors regardless of whether we got an error response or not